### PR TITLE
Gardening: Migrate test suite to GH issues p. 22

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

File: test/SourceKit/Indexing/issue-46400.swift.response [hunk 1]
```diff
@@ -11,22 +11,22 @@
     {
       key.kind: source.lang.swift.decl.protocol,
       key.name: "P",
-      key.usr: "s:7sr_38151PP",
-      key.line: 4,
+      key.usr: "s:4main1PP",
+      key.line: 6,
```
Comment (AnthonyLatsis) [inline test/SourceKit/Indexing/issue-46400.swift.response:15]: +2 lines due to the issue link in the original file.

File: test/SourceKit/SyntaxMapData/Inputs/syntaxmap.swift [hunk 2]
```diff
@@ -56,10 +56,10 @@ func testArgumentLabels(in class: Int, _ case: (_ default: Int) -> Void) -> (in:
   return something ? result : (in: 2, "foo")
 }
 
-// https://bugs.swift.org/browse/SR-9576
+// https://github.com/apple/swift/issues/52023
 func someFunc(input :Int?, completion: () throws -> Void) rethrows {}
 
-// https://bugs.swift.org/browse/SR-9740
+// https://github.com/apple/swift/issues/52171
```
Comment (AnthonyLatsis) [inline test/SourceKit/SyntaxMapData/Inputs/syntaxmap.swift:62]: This change broke the test due to a shift in the byte offsets of subsequent syntactic elements, so I had to update the response file too.

File: test/SourceKit/DocumentStructure/Inputs/main.swift [hunk 3]
```diff
@@ -105,10 +105,10 @@ protocol FooProtocol {
     associatedtype Baz: Equatable
 }
 
-// SR-5717
+// https://github.com/apple/swift/issues/48287
 a.b(c: d?.e?.f, h: i)
 
-// SR-6926
+// https://github.com/apple/swift/issues/49474
```
Comment (AnthonyLatsis) [inline test/SourceKit/DocumentStructure/Inputs/main.swift:111]: Ditto.

File: test/SILOptimizer/simplify_cfg_trivial_jumpthread.sil [hunk 4]
```diff
@@ -20,7 +20,9 @@ enum BoolLike { case true_, false_ }
 // Test that SimplifyCFG::simplifyBlocks, tryJumpThreading does not
 // perform unbounded loop peeling.
 //
-// rdar://73357726 ([SR-14068]: Compiling with optimisation runs indefinitely for grpc-swift)
+// rdar://73357726
+// [https://github.com/apple/swift/issues/56457
```
Comment (xedin) [inline test/SILOptimizer/simplify_cfg_trivial_jumpthread.sil:24]: Nit: Leftover `[`

**Review by xedin — APPROVED** [review]

File: test/SILOptimizer/simplify_cfg_trivial_jumpthread.sil (hunk 4 above)
Comment (AnthonyLatsis) [reply test/SILOptimizer/simplify_cfg_trivial_jumpthread.sil:24]: Well spotted!

**Comment by AnthonyLatsis** [comment]
@swift-ci please test

**Pull request merged**